src/cpp-common/include/cpp-common/HashMap.h

```diff
@@ -0,0 +1,34 @@
+#ifndef PROPOLIS_HASHMAP_H
+#define PROPOLIS_HASHMAP_H
+
+#include "IHashMap.h"
+#include <type_traits>
+
+template <typename Key, typename MappedType, uint16_t maxSize>
```

Comment:
You could do a base class that you inject the memory and it's size and then another class that wraps it and has a std:;aligned_storage and pass it at the constructor, like the CircualrQueue and CircularQueueStack. Though maybe that would be a bit tricky since you have two arrays, one for the bool and one for the data

Reply:
I prefer keeping everything in the template, will simplify usage

Reply:
It does change usage, you are just doing composition, so you could reuse the hashmap with any allocation, your array could come from a malloc, or a static array, or a std::aligned_storage

See https://github.com/SwarmUS/Propolis/blob/master/src/cpp-common/include/cpp-common/CircularQueueStack.h
and  https://github.com/SwarmUS/Propolis/blob/master/src/cpp-common/include/cpp-common/CircularQueue.tpp

Queue stack manages the memory, while Queue just handles the logic.

though I can see how it would be tricky. depends if you keep with an std::pair of a tuple with the bool in it, if it's a tuple wiht <bool, key, obj> it would be easier, if you want to manage it as 3 seperated array, you need to inject all three of the array. check the comment on the array vs tuple of 3